### ISSUE TITLE
Ember store selects incorrect model

### ISSUE THREAD

**Issue by AlexanderKapelyukhovskiy** [emberjs/data#5670]
I'm implementing new page for existing Ember application.
```
DEBUG: -------------------------------
DEBUG: Ember           : 2.16.4
DEBUG: Ember Data      : 2.16.4
DEBUG: jQuery          : 3.3.1
DEBUG: Model Fragments : 2.14.0
DEBUG: -------------------------------
```
I added new Model `call-info.js` and there already was `call.js`

After I tried to get data with help of store
```
model(params) {
    return this.store.findRecord('call-info', params.call_id);
}
```
I received object with only id attribute correctly set.

The problem is that store used call.js model for deserialization instead of call-info.js.

When I removed `call.js` model store returned correct object with all attributes correctly set.

Is there a way to fix this issue or somehow specify correct model ('call-info') for this.store.findRecord method?

**Comment by AlexanderKapelyukhovskiy** [comment]
I also was able to reproduce issue with `call ` model on newly created ember app

**Comment by AlexanderKapelyukhovskiy** [comment]
https://github.com/AlexanderKapelyukhovskiy/ember-call-model-issue

**Comment by runspired** [comment]
@AlexanderKapelyukhovskiy I don't see a reproduction in that repo... I see the setup but no mocking or stubbing of network requests.

**Comment by AlexanderKapelyukhovskiy** [comment]
@runspired I'm sorry I didn't provide api initially. When I started to create api I found source of the issue myself. https://github.com/AlexanderKapelyukhovskiy/ember-call-model-api-issue

Main problem was that api response contained property `type`
```
{
  calldetail: {
  ...
  type : call
  ...
  }
}
```
This field forced ember store to think that model was `call` but not `calldetail`. Once I removed this field issue gone.
